framework/src/utils/MultiMooseEnum.C

```diff
@@ -33,10 +33,7 @@ MultiMooseEnum::MultiMooseEnum(std::string names,
 }
 
 MultiMooseEnum::MultiMooseEnum(const MultiMooseEnum & other_enum)
-  : MooseEnumBase(other_enum),
-    _current_ids(other_enum._current_ids),
-    _current_names(other_enum._current_names),
-    _current_names_preserved(other_enum._current_names_preserved)
+  : MooseEnumBase(other_enum), _current(other_enum._current)
```

Comment:
This looks better. With MultiMooseEnum, the copy-constructor leaves the copy in a useable state as opposed to the one you implemented in MooseEnum.